src/modules/cpu/kernel/rotate.hpp

```diff
@@ -0,0 +1,1943 @@
+{
+    pSrcY = _mm_add_ps(pSrcY, pAffineMatrixTerm3Incr);   // Vectorized computation of next 4 src Y locations by adding the delta from previous location
+    pSrcX = _mm_add_ps(pSrcX, pAffineMatrixTerm0Incr);   // Vectorized computation of next 4 src X locations by adding the delta from previous location
+}
+
+inline void compute_rotate_src_loc_next_term_avx(__m256 &pSrcY, __m256 &pSrcX, __m256 &pAffineMatrixTerm3Incr, __m256 &pAffineMatrixTerm0Incr)
+{
+    pSrcY = _mm256_add_ps(pSrcY, pAffineMatrixTerm3Incr);   // Vectorized computation of next 8 src Y locations by adding the delta from previous location
+    pSrcX = _mm256_add_ps(pSrcX, pAffineMatrixTerm0Incr);   // Vectorized computation of next 8 src X locations by adding the delta from previous location
+}
+
+inline void compute_rotate_src_loc(Rpp32s dstY, Rpp32s dstX, Rpp32f &srcY, Rpp32f &srcX, Rpp32f6 &affineMatrix_f6, Rpp32s roiHalfHeight, Rpp32s roiHalfWidth)
+{
+    dstX -= roiHalfWidth;
+    dstY -= roiHalfHeight;
+    srcX = std::fma(dstX, affineMatrix_f6.data[0], std::fma(dstY, affineMatrix_f6.data[1], affineMatrix_f6.data[2])) + roiHalfWidth;
+    srcY = std::fma(dstX, affineMatrix_f6.data[3], std::fma(dstY, affineMatrix_f6.data[4], affineMatrix_f6.data[5])) + roiHalfHeight;
+}
+
+inline void compute_rotate_src_loc_next_term(Rpp32s dstX, Rpp32f &srcY, Rpp32f &srcX, Rpp32f6 &affineMatrix_f6)
+{
+    srcY += affineMatrix_f6.data[3];   // Computation of next src Y locations by adding the delta from previous location
+    srcX += affineMatrix_f6.data[0];   // Computation of next src X locations by adding the delta from previous location
+}
+
+/************* NEAREST NEIGHBOR INTERPOLATION *************/
+
+RppStatus rotate_nn_u8_u8_host_tensor(Rpp8u *srcPtr,
+                                           RpptDescPtr srcDescPtr,
+                                           Rpp8u *dstPtr,
+                                           RpptDescPtr dstDescPtr,
+                                           Rpp32f *angle,
+                                           RpptROIPtr roiTensorPtrSrc,
+                                           RpptRoiType roiType,
+                                           RppLayoutParams srcLayoutParams)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+
+omp_set_dynamic(0);
+#pragma omp parallel for num_threads(dstDescPtr->n)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi, roiLTRB;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+        compute_ltrb_from_xywh_host(&roi, &roiLTRB);
+        Rpp32s roiHalfWidth = roi.xywhROI.roiWidth >> 1;
+        Rpp32s roiHalfHeight = roi.xywhROI.roiHeight >> 1;
+        
+        Rpp32f6 affineMatrix_f6;
+        compute_rotate_affine_matrix(angle[batchCount], affineMatrix_f6);
```

Comment:
Just like the HIP, where you don't introduce a kernel file for rotate at all, can we have a quick rotate affine matrix computation in the API file itself for HOST too, and have the API file just call warp affine host tensor kernels? That would avoid all this code entirely since its same as warp affine. Pls check.

Reply:
That would introduce 2 for loops which iterate through the batch size, one for the affine matrix computation
The other for the warp affine computation. Hence thought of creating new file for HOST alone, please let me know if this is alright.

Reply:
that should be fine since it host, but check performance with and without an omp on the rotate affine matrix computation loop. So each thread would pretty much be responsible to compute just one sincos